### ISSUE TITLE
Default / Custom Discord Commands after Upgrade not accessible

### ISSUE THREAD

**Issue by wofnull** [PhantomBot/PhantomBot#3535]
PhantomBot Version: 3.14.0.6 ( using MariaDB / MYSQL )

O/S Version: 
Linux ( Debian ) 12.6

Java Version:
java version "21.0.3" 2024-04-16 LTS
Java(TM) SE Runtime Environment (build 21.0.3+7-LTS-152)
Java HotSpot(TM) 64-Bit Server VM (build 21.0.3+7-LTS-152, mixed mode, sharing)

Detailed Description:
After updating from Version 3.6.5.1, i cannot access the discord comand settings in the web panel.

There comes up an error upon trying to access custom or default comands:

`Processing Error
An exception was thrown while attempting to handle the request. See the core-error log for details`

After checking in the logs and the SQL server, it looks like at least one table is missing. 

The rest of the bot in twitch an discord is working as intended.

Logs:

```
[09-17-2024 @ 20:33:57.997 GMT] [handleUnrestrictedCommands()@WsPanelHandler.java:484] ***E***x***c***e***p***t***i***o***n*** ***p***r***o***c***e***s***s***i***n***g*** ***/***w***s***/***p***a***n***e***l*** ***f***r***a***m***e***:*** ***{***"***d***b***k***e***y***s***l***i***s***t***"***:***"***d***i***s***c***o***r***d***_***c***o***m***m***a***n***d***s***_***g***e***t***_***c***u***s***t***o***m***"***,***"***q***u***e***r***y***"***:***[***{***"***t***a***b***l***e***"***:***"***d***i***s***c***o***r***d***C***o***m***m***a***n***d***s***"***}***,***{***"***t***a***b***l***e***"***:***"***d***i***s***c***o***r***d***P***e***r***m***s***O***b***j***"***}***,***{***"***t***a***b***l***e***"***:***"***d***i***s***c***o***r***d***P***e***r***m***c***o***m***"***}***]***,***"***s***e***c***t***i***o***n***"***:***"***d***i***s***c***o***r***d***"***}***
[09-17-2024 @ 20:33:57.998 GMT] [handleUnrestrictedCommands()@WsPanelHandler.java:485] ***o***r***g***.***j***o***o***q***.***e***x***c***e***p***t***i***o***n***.***D***a***t***a***A***c***c***e***s***s***E***x***c***e***p***t***i***o***n***:*** ***S***Q***L*** ***[***s***e***l***e***c***t*** ***`***p***h***a***n***t***o***m***b***o***t***_***d***i***s***c***o***r***d***C***o***m***m***a***n***d***s***`***.***`***s***e***c***t***i***o***n***`***,*** ***`***p***h***a***n***t***o***m***b***o***t***_***d***i***s***c***o***r***d***C***o***m***m***a***n***d***s***`***.***`***v***a***r***i***a***b***l***e***`***,*** ***`***p***h***a***n***t***o***m***b***o***t***_***d***i***s***c***o***r***d***C***o***m***m***a***n***d***s***`***.***`***v***a***l***u***e***`*** ***f***r***o***m*** ***`***p***h***a***n***t***o***m***b***o***t***_***d***i***s***c***o***r***d***C***o***m***m***a***n***d***s***`*** ***w***h***e***r***e*** ***(***`***p***h***a***n***t***o***m***b***o***t***_***d***i***s***c***o***r***d***C***o***m***m***a***n***d***s***`***.***`***s***e***c***t***i***o***n***`*** ***=*** ***?*** ***a***n***d*** ***`***p***h***a***n***t***o***m***b***o***t***_***d***i***s***c***o***r***d***C***o***m***m***a***n***d***s***`***.***`***v***a***r***i***a***b***l***e***`*** ***=*** ***?***)***]***;*** ***(***c***o***n***n***=***1***6***9***9***7***3***)*** ***T***a***b***l***e*** ***'***p***h***a***n***t***o***m***-***w***o***f***n***u***l***l***.***p***h***a***n***t***o***m***b***o***t***_***d***i***s***c***o***r***d***C***o***m***m***a***n***d***s***'*** ***d***o***e***s***n***'***t*** ***e***x***i***s***t***
[09-17-2024 @ 20:33:57.999 GMT] [handleUnrestrictedCommands()@WsPanelHandler.java:485]
org.jooq.exception.DataAccessException: SQL [select `phantombot_discordCommands`.`section`, `phantombot_discordCommands`.`variable`, `phantombot_discordCommands`.`value` from `phantombot_discordCommands` where (`phantombot_discordCommands`.`section` = ? and `phantombot_discordCommands`.`variable` = ?)]; (conn=169973) Table 'phantom-wofnull.phantombot_discordCommands' doesn't exist
	at org.jooq_3.18.7.MARIADB.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3470)
	at org.jooq.impl.Tools.translate(Tools.java:3458)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:802)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:360)
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:301)
	at org.jooq.impl.SelectImpl.fetchLazy(SelectImpl.java:2857)
	at org.jooq.impl.DefaultDSLContext.fetchLazy(DefaultDSLContext.java:5195)
	at org.jooq.impl.DefaultDSLContext.fetchOne(DefaultDSLContext.java:4886)
	at org.jooq.impl.DefaultDSLContext.fetchOptional(DefaultDSLContext.java:5155)
	at org.jooq.impl.DefaultDSLContext.fetchOptional(DefaultDSLContext.java:5160)
	at com.gmt2001.datastore.DataStore.OptRecord(DataStore.java:704)
	at com.gmt2001.datastore.DataStore.OptRecord(DataStore.java:685)
	at com.gmt2001.datastore.DataStore.OptRecord(DataStore.java:672)
	at com.gmt2001.datastore.DataStore.GetString(DataStore.java:744)
	at tv.phantombot.panel.WsPanelHandler.handleDBKeysListQuery(WsPanelHandler.java:726)
	at tv.phantombot.panel.WsPanelHandler.handleUnrestrictedCommands(WsPanelHandler.java:473)
	at tv.phantombot.panel.WsPanelHandler.handleFrame(WsPanelHandler.java:113)
	at com.gmt2001.httpwsserver.WebSocketFrameHandler.channelRead0(WebSocketFrameHandler.java:88)
	at com.gmt2001.httpwsserver.WebSocketFrameHandler.channelRead0(WebSocketFrameHandler.java:46)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at io.netty.handler.codec.http.websocketx.Utf8FrameValidator.channelRead(Utf8FrameValidator.java:89)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1475)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1338)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1387)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at com.gmt2001.httpwsserver.RequestLogger.channelRead(RequestLogger.java:59)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:509)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.sql.SQLSyntaxErrorException: (conn=169973) Table 'phantom-wofnull.phantombot_discordCommands' doesn't exist
	at org.mariadb.jdbc.export.ExceptionFactory.createException(ExceptionFactory.java:289)
	at org.mariadb.jdbc.export.ExceptionFactory.create(ExceptionFactory.java:378)
	at org.mariadb.jdbc.message.ClientMessage.readPacket(ClientMessage.java:172)
	at org.mariadb.jdbc.client.impl.StandardClient.readPacket(StandardClient.java:886)
	at org.mariadb.jdbc.client.impl.StandardClient.readResults(StandardClient.java:825)
	at org.mariadb.jdbc.client.impl.StandardClient.readResponse(StandardClient.java:744)
	at org.mariadb.jdbc.client.impl.StandardClient.execute(StandardClient.java:668)
	at org.mariadb.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:91)
	at org.mariadb.jdbc.ClientPreparedStatement.execute(ClientPreparedStatement.java:270)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219)
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4761)
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:236)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:346)
	... 70 more
```

**Comment by gmt2001** [comment]
Did you have custom commands for Discord previously?

**Comment by wofnull** [comment]
there were none setup for discord from my end, I only used the standard commands, which are unavailable aswell ( same error ) since the update

**Comment by gmt2001** [comment]
Can you check what exact version of MariaDB is installed

**Comment by wofnull** [comment]
the current used version: 10.6.8-MariaDB-1:10.6.8

**Comment by gmt2001** [comment]
Should be fixed in next nightly

**Comment by wofnull** [comment]
Testing the nightly currently, it looks like the error itself is gone, but now my custom comands for the twitch bot are gone, aswell as the default commands from twitch and discord side of the bot ... these are now empty.

as far as i can see everything seems to be empty now:

- Event History
- saved / current title(s)
- permissions
- timers
- alerts
- channelpoints
- rankings
- history 
...
and so on ...

**Comment by gmt2001** [comment]
So the issue that I noticed and fixed was that only some parts of the database scripts were honoring the custom dbname you set (phantom-wofnull)

So you may need to migrate your data from the default dbname over to the one you want, or change the bot to just use the default

**Comment by wofnull** [comment]
It looks more as if the bot created completly new tables in the existing db, with changed upper / lowercase in some parts of the table name:
![image](https://github.com/user-attachments/assets/8dd83e8c-d9b9-4278-991b-de73dbf38efd)
this goes on and on over the whole DB

**Comment by gmt2001** [comment]
Check if 1 set of tables has the data and the other set is blank. If so, try dropping the blank tables

Then restart the bot

**Comment by gmt2001** [comment]
Actually wait. I need to change something first

**Comment by wofnull** [comment]
jep without change, the tables in lowercase will recreate on Bot start

**Comment by gmt2001** [comment]
New nightly build is out. Try that. Delete the blank tables again

**Comment by wofnull** [comment]
Bot shut down, tables deleted, bot updated to current Nightly, bot restarted, recreates lower case "new" tables with standardsettings, as before. as suspected, webinterface stays empty as before.

**Comment by gmt2001** [comment]
Very strange. It seems like it just can't see the old tables

**Comment by gmt2001** [comment]
Can you login to the database using the username and password that the bot uses and check that it can see the tables

**Comment by wofnull** [comment]
the user for the bot can see the database , all tables and can access them without issues.

the database credentials were not changed during the initial bot upgrade or later.

**Comment by gmt2001** [comment]
That's what I thought

**Comment by gmt2001** [comment]
I made a new nightly build

**Comment by wofnull** [comment]
I tested the new build, tables are not created this time, but during startup thebot throws many Errors
![image](https://github.com/user-attachments/assets/59077ab2-55f8-4f8b-af26-e94ddc44beb9)

Everything in the Panel is back as it was before the first nightly build, but aswell the discord commands problem came up again:
![image](https://github.com/user-attachments/assets/96d50194-17cf-4f39-98fb-76bc2e18e018)

**Comment by gmt2001** [comment]
Can you try executing `CHECK TABLE phantom-wofnull.phantombot_discordCommands` on the database

**Comment by wofnull** [comment]
As said initially: the table is missing from the database since the first update ... , therefore:
![image](https://github.com/user-attachments/assets/74b1377e-09c4-4891-afa8-4571161691f3)

only discord related Tables in the database:
![image](https://github.com/user-attachments/assets/ac9ee1b7-de9b-442b-9b37-1b73ac4a5821)

**Comment by gmt2001** [comment]
Lets try something else that I just pushed in a nightly

**Comment by wofnull** [comment]
Bot is now running with the latest nightly. it seems its stable for now, but the webinterface seems to be laggy as hell and needs a lot of time to load between pages. When this happens the bot throws an error , but after a few seconds it loads anyways.

However: The database has now the missing table and the discord commands section now opens without throwing errors, but the default discord commands are still missing ... if there were any.

I attach the core-errorlog from today.
[2024-09-21.txt](https://github.com/user-attachments/files/17084233/2024-09-21.txt)

**Comment by wofnull** [comment]
Further testing shows, the bot is joining the twitch chat, and additional to the delay in the webinterface, the bot stopped to react on any command  rendering the bot itself useless

**Comment by gmt2001** [comment]
Please try the latest nightly. I added 1 more change a few days ago

**Comment by wofnull** [comment]
Ok with the currently ngihtly, it seems to work normal again, bot reacts to chatcommands as it should, and the delay / errors in thte webinterface / logs are gone.

one last thing that is open: where there any default commands for the discord bot itself?, because this portion is still displayed empty.

**Comment by gmt2001** [comment]
For custom commands, no. There should be some built-in script commands if the modules are enabled

**Comment by wofnull** [comment]
The modules are enabled, but not displaying anything.

I did however run with the nightly in other DB related issues after trying to update some information in the bot itself:

1.) in some modules like greetings Module for specific People, the config for the Module appeared out of nowhere in the table for the users to greet. However I was able to remove those lines in the bot webinterface without affecting the settings itself

2.) in some modules ( like greeting / global ranks / personal ranks / and so on ) , trying to add new entrys or edit existing ones, result in not updating at all in webinterface or database, or ( which is worse ) in deleting random entrys in the specific table.

As the bot was running since 2018, and data loss seems now to be not a thing not to have, i have shut down the MariaDB Version of the bot and did a clean install with the current normal Build with the local DB instead. 

Would be nice to see the problems get fixxed in the future, but for now I think , my bot will run on SQL-Lite.
